cfme/tests/cloud_infra_common/test_html5_vm_console.py

```diff
@@ -42,18 +28,39 @@ def vm_obj(request, provider, setup_provider, console_template):
                                     template_name=console_template.name)
 
     request.addfinalizer(lambda: vm_obj.cleanup_on_provider())
-
     vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default")
-    if provider.one_of(OpenStackProvider):
-        # Assign FloatingIP to Openstack Instance from pool
-        # so that we can SSH to it
-        public_net = provider.data['public_network']
-        vm_obj.mgmt.assign_floating_ip(public_net)
     return vm_obj
 
 
+@pytest.mark.tier(2)
+@test_requirements.html5
+@pytest.mark.provider([VMwareProvider], selector=ONE)
+@pytest.mark.meta(automates=[BZ(1514594)])
+def test_html5_console_ports_present(appliance, setup_provider, provider):
+    """
+    Bugzilla:
+        1514594
+
+    Check to see if the Add/Edit provider screen has the Host VNC Start Port
+    and Host VNC End port. Only applicable to versions of VMware that support VNC console.
+
+    Polarion:
+        assignee: apagac
+        casecomponent: Appliance
+        initialEstimate: 1/4h
+        startsin: 5.8
+    """
+    edit_view = navigate_to(provider, 'Edit')
+    assert edit_view.vnc_start_port.is_displayed
+    assert edit_view.vnc_end_port.is_displayed
+
+
 @pytest.mark.rhv1
-def test_html5_vm_console(appliance, provider, configure_websocket, vm_obj,
+@test_requirements.html5
+@pytest.mark.provider(gen_func=providers,
```

Comment:
Wasn't trying to comment on every one of these - but same thing here, this is a basic use of provider marker, passing it a list of classes and required_flags.

Also the combination of module scoped provider marker and function scoped `setup_provider` fixture, which I don't think is intentional.